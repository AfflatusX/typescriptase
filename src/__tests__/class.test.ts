/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/tests.ts::classTest>>
 * BESPOKE<<imports, beforeAll, afterAll, beforeEach, afterEach, class1, class2, class3>>
 * SIGNED<<lQVIrPnCp92XZVAYwj+YlIBjw+6oIRmFh119H2jN4rR7z0vFRMZ4WfDL0B9GWbM41s2/MAUFb2kbDwinnuEjWw==>>
 */

/* BESPOKE START <<imports>> */
import { class1, class2, class3 } from "./examples";
/* BESPOKE END <<imports>> */

beforeAll(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeAll>> */
    /* BESPOKE END <<beforeAll>> */
  },
);

afterAll(
  async (): Promise<void> => {
    /* BESPOKE START <<afterAll>> */
    /* BESPOKE END <<afterAll>> */
  },
);

beforeEach(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeEach>> */
    /* BESPOKE END <<beforeEach>> */
  },
);

afterEach(
  async (): Promise<void> => {
    /* BESPOKE START <<afterEach>> */
    /* BESPOKE END <<afterEach>> */
  },
);

test(
  "class1",
  async (): Promise<void> => {
    /* BESPOKE START <<class1>> */
    expect(class1.print({
      name: "class1",
      path: "src/__tests__/class.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<class1>> */
  },
);

test(
  "class2",
  async (): Promise<void> => {
    /* BESPOKE START <<class2>> */
    expect(class2.print({
      name: "class2",
      path: "src/__tests__/class.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<class2>> */
  },
);

test(
  "class3",
  async (): Promise<void> => {
    /* BESPOKE START <<class3>> */
    expect(class3.print({
      name: "class3",
      path: "src/__tests__/class.test.ts",
    }))
      .toMatchSnapshot();
    /* BESPOKE END <<class3>> */
  },
);
