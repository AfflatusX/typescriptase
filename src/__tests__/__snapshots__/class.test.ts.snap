// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class1 1`] = `
"export abstract class MyClass1<T> {

  public propertyInstance1: string = \\"MYVAR\\";

  public static propertyStatic1: string = \\"MYVAR\\";

  public async methodInstance1<T>(
    var1: string,
    var2: object = {},
  ): Promise<string[]> {
  }

  public static async methodStatic1<T>(
    var1: string,
    var2: object = {},
  ): Promise<string[]> {
  }
}
"
`;

exports[`class2 1`] = `
"export class MyClass2<T, V> extends MyClass1 {

  /* BESPOKE START <<TEST1>> */
  /* BESPOKE END <<TEST1>> */

  protected readonly propertyInstance2: number;

  protected readonly static propertyStatic2: number;

  protected abstract methodInstance2<T, V>(
  ): void;

  protected static abstract methodStatic2<T, V>(
  ): void;
}
"
`;

exports[`class3 1`] = `
"class MyClass3 extends MyClass1 implements MyInterface1, MyInterface2 {

  private propertyInstance3: string | null;

  private static propertyStatic3: string | null;

  private async methodInstance3(
  ): void {
    /* BESPOKE START <<methodInstance3Bespoke>> */
    /* BESPOKE END <<methodInstance3Bespoke>> */
  }

  private static async methodStatic3(
  ): void {
    /* BESPOKE START <<methodStatic3Bespoke>> */
    /* BESPOKE END <<methodStatic3Bespoke>> */
  }
}
"
`;
